File: Talleres/LaboRobot_1/LuckySpawn.java
package aspitia.mattei;
import robocode.JuniorRobot;

public class LuckySpawn extends JuniorRobot
{
    BattleStrategy strategy;

    public LuckySpawn() {
        super();
        this.strategy = new GridWalker();
    }

	public void run() {
        this.setColors(black, blue, red, red, black);
        strategy.run(this);
	}

	public void onScannedRobot() {
        strategy.onScannedRobot(this);
	}

	public void onHitByBullet() {
        strategy.onHitByBullet(this);
	}

	public void onHitWall() {
        strategy.onHitWall(this);
	}

    public void onHitRobot() {
        strategy.onHitRobot(this);
    }
}

File: Talleres/LaboRobotKotlin_1/LuckySpawn.kt
package aspitia.mattei.kotlin

import robocode.JuniorRobot


class LuckySpawn : JuniorRobot() {
    private var strategy: BattleStrategy = GridWalker()

    override fun run() {
        this.setColors(black, blue, red, red, black)
        strategy.run(this)
    }

    override fun onScannedRobot() {
        strategy.onScannedRobot(this)
    }

    override fun onHitByBullet() {
        strategy.onHitByBullet(this)
    }

    override fun onHitWall() {
        strategy.onHitWall(this)
    }

    override fun onHitRobot() {
        strategy.onHitRobot(this)
    }
}